otcextensions/tests/unit/osclient/nat/v2/test_gateway.py

```diff
@@ -0,0 +1,185 @@
+            (s.id, s.name, s.spec, s.router_id, s.status))
+
+    def setUp(self):
+        super(TestListNatGateways, self).setUp()
+
+        self.cmd = gateway.ListNatGateways(self.app, None)
+
+        self.client.gateways = mock.Mock()
+        self.client.api_mock = self.client.gateways
+
+    def test_list(self):
+        arglist = []
+
+        verifylist = []
+
+        # Verify cm is triggered with default parameters
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        # Set the response
+        self.client.api_mock.side_effect = [self.objects]
+
+        # Trigger the action
+        columns, data = self.cmd.take_action(parsed_args)
+
+        self.client.api_mock.assert_called_with()
+
+        self.assertEqual(self.column_list_headers, columns)
+        self.assertEqual(self.data, list(data))
+
+    def test_list_args(self):
+        arglist = [
+            '--limit', '1',
+            '--id', '2',
+            '--name', '3',
+            '--project-id', '4',
+            '--spec', '5',
+            '--router-id', '6',
+            '--internal-network-id', '7',
+            '--admin-state-up', '8',
+            '--created-at', '9',
+            '--status', '10'
+        ]
+
+        verifylist = [
+            ('limit', 1),
+            ('id', '2'),
+            ('name', '3'),
+            ('project_id', '4'),
+            ('spec', '5'),
+            ('router_id', '6'),
+            ('internal_network_id', '7'),
+            ('admin_state_up', '8'),
+            ('created_at', '9'),
+            ('status', '10'),
+        ]
+
+        # Verify cm is triggered with default parameters
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        # Set the response
+        self.client.api_mock.side_effect = [self.objects]
+
+        # Trigger the action
+        columns, data = self.cmd.take_action(parsed_args)
+
+        self.client.api_mock.assert_called_with(
+            limit=1,
+            id='2',
+            name='3',
+            project_id='4',
+            spec='5',
+            router_id='6',
+            internal_network_id='7',
+            admin_state_up='8',
+            created_at='9',
+            status='10',
+        )
+
+
+class TestShowNatGateway(fakes.TestNat):
+
+    _data = fakes.FakeNatGateway.create_one()
+
+    columns = (
+        'admin_state_up',
+        'created_at',
+        'description',
+        'id',
+        'internal_network_id',
+        'name',
+        'project_id',
+        'router_id',
+        'spec',
+        'status'
+    )
+
+    data = fakes.gen_data(_data, columns)
+
+    def setUp(self):
+        super(TestShowNatGateway, self).setUp()
+
+        self.cmd = gateway.ShowNatGateway(self.app, None)
+
+        self.client.find_gateway = mock.Mock(return_value=self._data)
+
+    def test_show(self):
+        arglist = [
+            'test_gateway',
+        ]
+
+        verifylist = [
+            ('gateway', 'test_gateway'),
```

Comment:
why you have gateway in verifylist? In arglist you have only test_gateway

Reply:
because gateway is positional argument
gateway is arg name and test_gateway is value.